fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/api/world/poi/v1/PointOfInterestHelper.java

```diff
@@ -0,0 +1,69 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.world.poi.v1;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+import com.google.common.collect.ImmutableSet;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.poi.PointOfInterestType;
+
+import net.fabricmc.fabric.mixin.object.builder.PointOfInterestTypeAccessor;
+
+/**
+ * This class provides utilities to create a {@link PointOfInterestType}.
+ */
+public interface PointOfInterestHelper {
+	/**
+	 * Registers and creates a {@link PointOfInterestType}.
+	 * @param workStationStates A list of {@link BlockState BlockStates} which this {@link PointOfInterestType} can be present on.
+	 * @param id The id of this {@link PointOfInterestType}.
+	 * @param ticketCount The amount of tickets.
+	 * @param searchDistance The search distance.
+	 * @return a new {@link PointOfInterestType}.
+	 */
+	static PointOfInterestType create(Set<BlockState> workStationStates, Identifier id, int ticketCount, int searchDistance) {
+		return Registry.POINT_OF_INTEREST_TYPE.add(id, PointOfInterestTypeAccessor.callSetup(PointOfInterestTypeAccessor.callCreate(id.toString(), workStationStates, ticketCount, searchDistance)));
+	}
+
+	/**
+	 * Registers and creates a {@link PointOfInterestType}.
+	 * @param workStationStates A list of {@link BlockState BlockStates} which this {@link PointOfInterestType} can be present on.
+	 * @param id The id of this {@link PointOfInterestType}.
+	 * @param ticketCount The amount of tickets.
+	 * @param completionCondition A {@link Predicate} which determines whether this point of interest type should be present.
+	 * @param searchDistance The search distance.
+	 * @return a new {@link PointOfInterestType}.
+	 */
+	static PointOfInterestType register(Set<BlockState> workStationStates, Identifier id, int ticketCount, Predicate<PointOfInterestType> completionCondition, int searchDistance) {
+		return Registry.POINT_OF_INTEREST_TYPE.add(id, PointOfInterestTypeAccessor.callSetup(PointOfInterestTypeAccessor.callCreate(id.toString(), workStationStates, ticketCount, completionCondition, searchDistance)));
+	}
+
+	/**
+	 * Gets all possible {@link BlockState BlockStates} a Block could have.
+	 * @param block The block to get all states from.
+	 * @return All possible {@link BlockState BlockStates} the block could contain.
+	 */
+	static Set<BlockState> getAllStatesOf(Block block) {
```

Comment:
It's more of a utility method for part of the setup for a point of interest type since it is present based on the state of the block.

It's more related to workStationStates in the method above.

Reply:
Just add a variant of your method that accepts a block instance then.